### PR TITLE
[Ubuntu] Move the list of installed Java versions to toolset 

### PR DESCRIPTION
**Pull request by nikita-bykov**
# Description
Moved the list of installed Java versions to toolset on Ubuntu

#### Related issue:
[#1635](https://github.com/actions/virtual-environments-internal/issues/1635)
## Check list
- [x] Related issue / work item is attached
- [x] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/linux/scripts/installers/java-tools.sh [hunk 1]
```diff
@@ -24,11 +28,8 @@ if isUbuntu16 || isUbuntu18 ; then
     apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 0xB1998361219BD9C9
     apt-add-repository "deb https://repos.azul.com/azure-only/zulu/apt stable main"
     apt-get update
-    apt-get -y install zulu-7-azure-jdk=\*
     # Open JDP Adopt does not exist for Ubuntu 20
-    apt-get -y install adoptopenjdk-12-hotspot=\*
-    echo "JAVA_HOME_7_X64=/usr/lib/jvm/zulu-7-azure-amd64" | tee -a /etc/environment
-    DEFAULT_JDK_VERSION=8
+    DEFAULT_JDK_VERSION=$JAVA_DEFAULT
```
Comment (maxim-lobanov) [inline images/linux/scripts/installers/java-tools.sh:32]: Why do we need two separate variables `DEFAULT_JDK_VERSION` and `JAVA_DEFAULT`?

File: images/linux/scripts/installers/java-tools.sh [hunk 2]
```diff
@@ -24,11 +28,8 @@ if isUbuntu16 || isUbuntu18 ; then
...
     defaultLabel8="(default)"
```
Comment (maxim-lobanov) [inline images/linux/scripts/installers/java-tools.sh:33]: We should set this label dynamically without hardcode too

File: images/linux/scripts/installers/java-tools.sh [hunk 3]
```diff
@@ -38,9 +39,16 @@ if isUbuntu20 ; then
     apt-get update
 fi
 
-# Install only LTS versions.
-apt-get -y install adoptopenjdk-8-hotspot=\*
-apt-get -y install adoptopenjdk-11-hotspot=\*
+for JAVA_VERSION in "${JAVA_VERSIONS_LIST[@]}"
```
Comment (maxim-lobanov) [inline images/linux/scripts/installers/java-tools.sh:42]: Why `DEFAULT_JDK_VERSION=11` is still hardcoded above?
We should get rid of both conditions `isUbuntu20` and `isUbuntu16 || isUbuntu18`

File: images/linux/scripts/installers/java-tools.sh [hunk 4]
```diff
@@ -38,9 +39,16 @@ if isUbuntu20 ; then
...
+    do
```
Comment (maxim-lobanov) [inline images/linux/scripts/installers/java-tools.sh:43]: `do` should be on previous line

File: images/linux/toolsets/toolset-2004.json [hunk 5]
```diff
@@ -59,6 +59,12 @@
             ]
         }
     ],
+    "java": {
+        "default": "11",
```
Comment (maxim-lobanov) [inline images/linux/toolsets/toolset-2004.json:63]: What about tests? I think we should change them too.

File: images/linux/scripts/installers/java-tools.sh [hunk 6]
```diff
@@ -15,6 +15,10 @@ function javaTool {
     fi
 }
 
+toolset="$INSTALLER_SCRIPT_FOLDER/toolset.json"
+JAVA_VERSIONS_LIST=$(jq -r '.java.versions | .[]' $toolset)
+JAVA_DEFAULT=$(jq -r '.java.default' $toolset)
```
Comment (al-cheb) [inline images/linux/scripts/installers/java-tools.sh:20]: nitpick: we don't use `JAVA_DEFAULT` var anywhere, I think we should rename to `DEFAULT_JDK_VERSION`.

File: images/linux/scripts/installers/java-tools.sh [hunk 7]
```diff
@@ -38,9 +39,16 @@ if isUbuntu20 ; then
     apt-get update
```
Comment (al-cheb) [inline images/linux/scripts/installers/java-tools.sh:39]: Where do we use defaultLabel11 and defaultLabel8?

File: images/linux/scripts/installers/java-tools.sh (hunk 1 above)
Comment (al-cheb) [inline images/linux/scripts/installers/java-tools.sh:32]: I think we can remove from this block this variable. Pls pay attention at the line 37.

**Comment by al-cheb** [comment]
Could you please pull the latest changes from main branch and fix comments?

File: images/linux/scripts/installers/java-tools.sh [hunk 8]
```diff
@@ -7,32 +7,28 @@
...
-    apt-add-repository "deb https://repos.azul.com/azure-only/zulu/apt stable main"
-    apt-get update
-    apt-get -y install zulu-7-azure-jdk=\*
-    # Open JDP Adopt does not exist for Ubuntu 20
-    apt-get -y install adoptopenjdk-12-hotspot=\*
-    echo "JAVA_HOME_7_X64=/usr/lib/jvm/zulu-7-azure-amd64" | tee -a /etc/environment
-    DEFAULT_JDK_VERSION=8
-    defaultLabel8="(default)"
-fi
+# Install GPG Key for Azul Open JDK. See https://www.azul.com/downloads/azure-only/zulu/
```
Comment (maxim-lobanov) [inline images/linux/scripts/installers/java-tools.sh:18]: let's return condition `if isUbuntu16 || isUbuntu18 ; then`

**Review by J-Vin — APPROVED** [review]

**Review by al-cheb — APPROVED** [review]

**Review by maxim-lobanov — APPROVED** [review]

**Pull request merged**